plugins/pulp_rpm/plugins/migrations/0037_rpm_primary_repodata_new_location.py

```diff
@@ -0,0 +1,171 @@
+from cStringIO import StringIO
+import sys
+import os
+
+from pulp.server.db import connection
+
+if sys.version < (2, 7):
+    # We need the non-C implementation so we can shove a namespace into its internal data structure,
+    # since in python 2.6, the register_namespace function wasn't present.
+    import xml.etree.ElementTree as ET
+else:
+    import xml.etree.cElementTree as ET
+
+
+RPM_NAMESPACE = 'http://linux.duke.edu/metadata/rpm'
+PULP_PACKAGES_DIR = "Packages"
+
+# this is required because some of the pre-migration XML tags use the "rpm"
+# namespace, which causes a parse error if that namespace isn't declared.
+FAKE_XML = '<?xml version="1.0" encoding="%(encoding)s"?><faketag ' \
+           'xmlns:rpm="%(namespace)s">%(xml)s</faketag>'
```

Comment:
You also have to [register the namespace](https://github.com/BrnoPCmaniak/pulp_rpm/blob/98455808642dedc01d69c11d8b6a80a9707a5d03/plugins/pulp_rpm/plugins/migrations/0033_checksums_and_templates.py#L33-L37) otherwise it will be `ns0`, check the `repodata` in `units_rpm` collection after migration.